### DISCUSSION TITLE
Why invalid enumeration value throws during parsing?

### DISCUSSION THREAD

**Discussion by tarsh** [ndsev/zserio#602] [Q&A]
Hello zserio,

I found a issue when using zserio.
If i define a enum object in V1 as this:
enum {
a,
b,
c,
};

but when in the V2 add new enum value as :
enum {
a,
b,
c,
d,
};

so if the client SDK (generate code by V1) to decode the new version V2 data, will throw a issue to make the application crash.
but as we needed is the all old application should to access the new data version to make the compatibility between of the application and databases.
if there any solution to solve it?

<img width="822" alt="image" src="https://github.com/ndsev/zserio/assets/834062/5d03098d-1caf-4f11-a56b-7956b7a6c2bd">

Thanks,

Best Regards！

tarsh dong

**Comment by mikir** [comment]
Hi Tarsh,

thanks for sharing your problem with us.

I am afraid that such forward compatibility of enumerations (meaning that old application should accept new version of enumeration) is not currently possible in zserio.

The problem with forward compatibility is that if we allow parsing of invalid enumeration value, we should somehow define invalid value in enumeration. This is because we cannot just leave uninialized enumeration value. Something like this:

```
enum
{
  a,
  b,
  c,
  i [[invalid]]
};
```

We will keep this problem in mind.

Btw, we are going to improve very soon backward compatibility, please see an issue #486.
